src/plugins/data/common/datasets/types.ts

```diff
@@ -247,6 +247,8 @@ export interface Dataset extends BaseDataset {
   timeFieldName?: string;
   /** Optional language to default to from the language selector */
   language?: string;
+  /** Optional name of the indexed view to search data from */
+  indexedView?: string;
```

Comment:
could you add more insight on what this is?
is this to be the id of the indexedView component. is this a unique id generated from something?

cc: @virajsanghvi curious your opinion here

without much context this for me this seems something like a data view is that correct @riysaxen-amzn ?